emma-language/src/main/scala/org/emmalanguage/io/csv/CSVColumn.scala

```diff
@@ -0,0 +1,132 @@
+
+  /** Creates a new codec for type `T` from pure conversion functions. */
+  def make[T](fromF: String => T, toF: T => String = (_: T).toString): CSVColumn[T] =
+    new CSVColumn[T] {
+      def from(column: String)(implicit csv: CSV) = fromF(column)
+      def to(value: T)(implicit csv: CSV) = toF(value)
+    }
+
+  /**
+   * Derives a new codec for type `B` from an existing codec for type `A`,
+   * given an isomorphism between `A` and `B`.
+   */
+  def iso[A, B](fromF: A => B, toF: B => A)(implicit A: CSVColumn[A]): CSVColumn[B] =
+    new CSVColumn[B] {
+      def from(column: String)(implicit csv: CSV) = fromF(A from column)
+      def to(value: B)(implicit csv: CSV) = A to toF(value)
+    }
+
+  // Scala primitives
+  implicit val stringCSVColumn:     CSVColumn[String]     = make(identity)
+  implicit val charCSVColumn:       CSVColumn[Char]       = make(_.head)
+  implicit val booleanCSVColumn:    CSVColumn[Boolean]    = make(_.toBoolean)
+  implicit val byteCSVColumn:       CSVColumn[Byte]       = make(_.toByte)
+  implicit val shortCSVColumn:      CSVColumn[Short]      = make(_.toShort)
+  implicit val intCSVColumn:        CSVColumn[Int]        = make(_.toInt)
+  implicit val longCSVColumn:       CSVColumn[Long]       = make(_.toLong)
+  implicit val floatCSVColumn:      CSVColumn[Float]      = make(_.toFloat)
+  implicit val doubleCSVColumn:     CSVColumn[Double]     = make(_.toDouble)
+  implicit val unitCSVColumn:       CSVColumn[Unit]       = make(const(()), const(""))
+  implicit val bigIntCSVColumn:     CSVColumn[BigInt]     = make(BigInt.apply)
+  implicit val bigDecimalCSVColumn: CSVColumn[BigDecimal] = make(BigDecimal.apply)
+
+  // Java primitives
+  implicit val jCharacterCSVColumn:  CSVColumn[jl.Character]  = make(_.head)
+  implicit val jBooleanCSVColumn:    CSVColumn[jl.Boolean]    = make(_.toBoolean)
+  implicit val jByteCSVColumn:       CSVColumn[jl.Byte]       = make(_.toByte)
+  implicit val jShortCSVColumn:      CSVColumn[jl.Short]      = make(_.toShort)
+  implicit val jIntegerCSVColumn:    CSVColumn[jl.Integer]    = make(_.toInt)
+  implicit val jLongCSVColumn:       CSVColumn[jl.Long]       = make(_.toLong)
+  implicit val jFloatCSVColumn:      CSVColumn[jl.Float]      = make(_.toFloat)
+  implicit val jDoubleCSVColumn:     CSVColumn[jl.Double]     = make(_.toDouble)
+  implicit val jBigIntegerCSVColumn: CSVColumn[jm.BigInteger] = make(new jm.BigInteger(_))
+  implicit val jBigDecimalCSVColumn: CSVColumn[jm.BigDecimal] = make(new jm.BigDecimal(_))
+  implicit val jDateCSVColumn:       CSVColumn[Date] = iso[Long, Date](new Date(_), _.getTime)
+
+  /**
+   * Creates a Scala [[Enumeration]] codec.
+   * Since enumerations are usually defined as objects,
+   * the best way to provide an implicit instance is via
+   * `implicit object MyEnum extends Enumeration`.
+   */
+  implicit def enumCSVColumn[E <: Enumeration](implicit enum: E): CSVColumn[enum.Value] =
```

Comment:
Enums in Scala are weird like that. It will be summoned correctly if we have an implicit instance of the enumeration (we can define it as an `implicit object` as the comment says):
```scala
implicit object ModelType extends Enumeration {
    val Multinomial = Value("multinomial")
    val Bernoulli = Value("bernoulli")
  }
```